src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -1488,6 +1489,161 @@ void addIfNotNull(INamedTypeSymbol? toAdd)
             ImmutableArray<bool> memberIsReadOnly,
             ImmutableArray<NullableAnnotation> memberNullableAnnotations);
 
+        /// <summary>
+        /// Creates an <see cref="IMethodSymbol"/> whose <see cref="IMethodSymbol.MethodKind"/> is <see
+        /// cref="MethodKind.BuiltinOperator"/> for a binary operator. Built-in operators are commonly created for
+        /// symbols like <c>bool int.operator==(int v1, int v2)</c> which the language implicitly supports, even if such
+        /// a symbol is not explicitly defined for that type in either source or metadata.
+        /// </summary>
+        /// <param name="name">The binary operator name.  Should be one of the names from <see cref="WellKnownMemberNames"/>.</param>
+        /// <param name="returnType">The return type of the binary operator.</param>
+        /// <param name="leftType">The type of the left operand of the binary operator.</param>
+        /// <param name="rightType">The type of the right operand of the binary operator.</param>
+        /// <param name="isChecked">Whether or not this is a checked binary operator.  For example <c>int int.operator
+        /// checked +(int v1, int v2)</c>.</param>
+        /// <returns></returns>
+        public IMethodSymbol CreateBuiltinOperator(string name, ITypeSymbol returnType, ITypeSymbol leftType, ITypeSymbol rightType, bool isChecked)
```

Comment:
>isChecked [](http://example.com/codeflow?start=138&length=9)

This parameter feels redundant since it is now part of the name. Or am my missing something?

Reply:
so, it seems like in VB this parameter is provided, indepedent of the name (unless i misunderstood the code).  For example:

https://github.com/dotnet/roslyn/blob/8c927833f64f13cc3ff2650e6cf64b0144c94c3d/src/Compilers/VisualBasic/Portable/BoundTree/BoundBinaryOperator.vb#L45-L55

Here the 'checked' flag is determined with some boolean checks of the types.  But the name is chosen independently.  The name-choosing code does this:

https://github.com/dotnet/roslyn/blob/8c927833f64f13cc3ff2650e6cf64b0144c94c3d/src/Compilers/VisualBasic/Portable/Semantics/Operators.vb#L217-L267

Which does not return the "checked" names at all.

Reply:
So i was trying to fit with that aspect for how VB/C# do things.  If we think that can be adjusted, i'm ok with it.  For example, we could have hte name make this determination, but hten in VB map the "checked" name to the "unchecked" form when actually making the method.

However this will make *using* this API much harder.  For example, if i have a VB checked built-in operator, it will give the name back as WellKnownMemberNames.AdditionOperatorName.  But if then need to *call* this api i'll have to pass along CheckedAdditionalOperatorName.  

I reallllllly wanted consumption and production of these symbols to be really easy.  The IDE is already being passed these symbols, we'd like to just grab the data we need off of them and then use that data to construct the symbols without having to worry about things potentially blowing up.

Reply:
> it seems like in VB this parameter is provided, indepedent of the name

Perhaps we should change that, and "teach" VB to take advantage of new names that are now part of the official set

Reply:
will do.  Based on offline conversation, i'll make a dedicated PR for this that we can review first.